src/index.ts

```diff
@@ -139,3 +138,110 @@ export class DynamoDBAutoIncrement extends BaseDynamoDBAutoIncrement {
     return { puts, nextCounter }
   }
 }
+
+/**
+ * Update a history table with an auto-incrementing attribute value in DynamoDB
+ *
+ * @example
+ * ```
+ * import { DynamoDB } from '@aws-sdk/client-dynamodb'
+ * import { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+ * import { DynamoDBHistoryAutoIncrement } from '@nasa-gcn/dynamodb-autoincrement'
+ *
+ * const client = new DynamoDB({})
+ * const doc = DynamoDBDocument.from(client)
+ *
+ * const autoIncrementHistory = DynamoDBHistoryAutoIncrement({
+ *   doc,
+ *   counterTableName: 'autoincrementHistoryTable',
+ *   counterTableKey: {
+ *     autoincrementItemID: 1
+ *   },
+ *   attributeName: 'version',
+ *   tableName: 'widgets',
+ *   initialValue: 1,
+ * })
+ *
+ * const latestVersionValue = await autoIncrementHistory.put({
+ *   widgetName: 'A new name for this item',
+ *   costDollars: 199.99,
+ * })
+ * ```
+ */
+export class DynamoDBHistoryAutoIncrement extends BaseDynamoDBAutoIncrement {
+  async #getLast(): Promise<number | undefined> {
+    return (
+      (
+        await this.props.doc.get({
+          AttributesToGet: [this.props.attributeName],
+          Key: this.props.counterTableKey,
+          TableName: this.props.tableName,
+        })
+      ).Item?.[this.props.attributeName] ?? undefined
+    )
+  }
+
+  protected async next(item: Record<string, NativeAttributeValue>) {
+    const counter = await this.#getLast()
+
+    let nextCounter
+
+    const existingUntrackedEntry = (
+      await this.props.doc.get({
+        TableName: this.props.tableName,
+        Key: this.props.counterTableKey,
+      })
+    ).Item
```

Comment:
No need to get the last item twice...

```suggestion
  protected async next(item: Record<string, NativeAttributeValue>) {
    let nextCounter

    const existingUntrackedEntry = (
      await this.props.doc.get({
        TableName: this.props.tableName,
        Key: this.props.counterTableKey,
      })
    ).Item

    const counter: number | undefined = existingUntrackedEntry?.[this.props.attributeName]
```